examples/gno.land/r/demo/gnolotto_factory/gnolotto_factory.gno

```diff
@@ -0,0 +1,229 @@
+package gnolotto_factory
+
+import (
+	"bytes"
+	"std"
+	"time"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/gnolotto"
+	"gno.land/p/demo/ufmt"
+	"gno.land/p/demo/seqid"
+)
+
+var lotteries *avl.Tree
+var lotteriesID seqid.ID
+
+// Replace this address with your address
+var admin std.Address = "g1jg8mtutu9khhfwc4nxmuhcpftf0pajdhfvsqf5"
+
+const TICKET_PRICE = 10
+
+// Initializes the lottery AVL tree
+func init() {
+	lotteries = avl.NewTree()
+}
+
+// Creates a new lottery, only callable by admin.
+func CreateLottery(drawTime int64, prizePool int64) (string, string) {
+	sentCoins := std.GetOrigSend()
+	amount := sentCoins.AmountOf("ugnot")
+
+	if prizePool != amount {
+		panic("Prize pool must match the transaction value")
+	}
+
+	if drawTime < time.Now().Unix() {
+		panic("Invalid draw time")
+	}
+
+	if std.GetOrigCaller() != admin {
+		panic("Only the admin can create a lottery")
+	}
+
+	lotteryID := lotteriesID.Next()
+	lottery := gnolotto.NewLottery(time.Unix(drawTime, 0), prizePool)
+
+	lotteries.Set(lotteryID.Binary(), lottery)
+	return lotteryID.String(), "Lottery created successfully"
+}
+
+// Buy ticket for a specific lottery.
+func BuyTicket(lotteryIDStr string, numbersStr string) (int, string) {
```

Comment:
We will hopefully soon [merge a PR](https://github.com/gnolang/gno/pull/1776) that will allow you to actually pass in slices as arguments to maketx call and other calls, which will help you avoid having to do this string parsing. If the lottery PR is merged before (and I assume it will be), please make a new PR for implement these changes once #1776 is merged :)